tests/unit/src/test/scala/tests/debug/DotEnvFileParserSuite.scala

```diff
@@ -81,11 +88,8 @@ class DotEnvFileParserSuite extends BaseSuite {
     assertDiffEqual(parse(" \nKEY=value"), keyValue)
   }
 
-  test("allow # and spaces in unquoted values") {
-    assertDiffEqual(
-      parse("KEY=v a l u e # not a comment"),
-      Map("KEY" -> "v a l u e # not a comment"),
-    )
```

Comment:
`KEY=v a l u e` is already nonconforming in shell as it tries to invoke `a` with env `KEY=v` and arguments of `l` `u` `e`. Arguably there are not much dialects supporting such cases at the cost of ambiguity.